oai_agents/common/teammates_collection.py

```diff
@@ -1,97 +1,190 @@
 from oai_agents.agents.rl import RLAgentTrainer
-from oai_agents.common.tags import TeamType, TeammatesCollection, AdversaryPlayConfig
+from oai_agents.common.tags import AgentPerformance, TeamType, TeammatesCollection, AdversaryPlayConfig
 
 from itertools import permutations
 import random
 from pathlib import Path
 
 
-def get_teammates(args, agents_perftag_score:list, teamtypes:list, teammates_len:int, unseen_teammates_len:int, agent:RLAgentTrainer=None):
+def get_teammates(agents_perftag_score:list, teamtypes:list, teammates_len:int, unseen_teammates_len:int, agent:RLAgentTrainer=None, use_entire_population:bool=False):
+    '''
+    Get the teammates for an agent to populate a TeammatesCollection
+
+    :param agents_perftag_score: The population of agents to select from represented as a list of tuples (agent, performance tag, score for this layout)
+    :param teamtypes: The list of TeamTypes to use to select teammates for each team
+    :param teammates_len: Total number of players on each team
+    :param unseen_teammates_len: For NXSP experiments, this is X, the number of unseen players that an SP agent is playing with
+    :param agent: The agent that the teammates will play with
+    :param use_entire_population: Flag inidicating if the entire population should be used, if False, this function will only generate one team for each teamtype
+    :returns: Dictionary of form `{<TeamType> : [[team1], [team2], ... ] ...}` and a list of all the agents that were selected for a team
+    '''
+
+    if use_entire_population:
+        # If we want to use the entire population, we must check that the population is evenly divisible by the number of required agents
+        required_population_size = 0
+        for team_type in teamtypes:
+            if team_type in TeamType.ALL_TYPES_BESIDES_SP:
+                required_population_size += (teammates_len - 1)
```

Comment:
@avv-va Follow up question on this... does it make sense to you if I exclude SELF_PLAY_ADVERSARY from the consideration since adversaries are added to the TC later on?

Reply:
Yes